README.md

```diff
@@ -4,7 +4,71 @@
 
 ## What is Tonatona?
 
-`Tonatona` is **meta** web application framework, which handles lots of annoying and boring tasks on real world web development such as environment variables, logging, etc...
+Tonatona is **meta** web application framework. It handles lots of boring tasks
+that are needed for real-world web development such as reading in values
+defined in environment variables, setting up logging, sending emails, accessing
+databases, etc.
 
-`Tonatona` **does not** provide explicit ways to realize core functionalities of web applications.
-It means that you can use your favorite framework based on `Network.Wai`! (e.g., Servant, Spock, Scotty, etc...)
+Tonatona **does not** provide the core functionalities of web applications,
+such as routing, request parsing, response building, etc.  Instead, you can
+use Tonatona with your favorite web framework.  Tonatona should work with Servant,
+Spock, Yesod, and others.
+
+Tonatona provides a plugin architecture so that anyone can add plugins
+implementing arbitrary functionality.  This repository contains many standard
+plugins that are helpful when writing Haskell applications.
+
+## How to use Tonatona
+
+(TODO)
+
+## Available Plugins
+
+(TODO)
+
+## Goals for Tonatona
+
+Tonatona is meant to make it easy to write web applications.  Its goal is to
+free the end-user from having to write anything but their application's
+business logic.
+
+We have the following goals and objectives in mind when developing Tonatona and
+the plugins.
+
+-   Tonatona should make it easy to get up and running with writing web
+    applications.  It is perfect when you are on a small team.
+
+    When you team or project grows larger, it may be worthwhile to move away
```

Comment:
I think Tonatona is probably suitable for larger projects, but in my experience, the larger a project gets, the more likely it is to need special customizations.  Once a project gets very large, I think it is more likely that the team will rip out Tonatona and hand-code the configuration parsing, logging, db access, email sending, etc.  (Although, this may not happen because of Tonatona's ability to add new plugins and functionality.)

I think it is more correct to say that Tonatona will work on almost any project, with any size of a team.  However, I am somewhat worried that Tonatona won't be adopted in the Haskell community if it doesn't have a smaller focus.  Branding Tonatona as a meta web application framework for small projects is a relatively narrow focus, but branding it as an application framework for any kind of project is a wide focus.  I think it would be easier to get people to adopt Tonatona at first if it has a narrow focus.

However, this is completely my opinion and may not be true at all!

Should I go ahead and change the explanation to say that Tonatona is suitable for any size of team/project?

Reply:
It makes sense!
So, how about saying "Tonatona focuses on typical applications"?
It is more straightforward than "focuses on small applications" and also has a narrow focus.

Reply:
I ended up taking this point out, because it doesn't really fit under "Features", and I rewrote the goals section.

Should something like this be added in somewhere?

Reply:
I don't know, to be honest.
So, how about submiting this repository to reddit after publishing to hackage?
If someone comments that "The target of this framework is too wide" or something similar, then rewrite the README.md and write a blog post introducing tonatona?